p4-16/psa/PSA.mdk

```diff
@@ -1073,28 +1072,26 @@ the contents of several metadata fields in the struct
 
     if (ostd.clone) {
         if (ostd.clone_session_id value is supported) {
-            cos = class_of_service configured for ostd.clone_session_id in PRE
-	    set((egress_port[0], instance[0]), .., (egress_port[n], instance[n])) =
-	        set of egress_port and instance pair in the
-	    	configured sessions for ostd.clone_session_id in PRE
-            trunc = truncate configured for ostd.clone_session_id in PRE
-            plen = packet_length_bytes configured for ostd.clone_session_id in PRE
+            from the values configured for ostd.clone_session_id in PRE {
+                cos = class_of_service
+                set((egress_port[0], instance[0]), ..., (egress_port[n], instance[n])) =
+                    set of egress_port and instance pairs
+                trunc = truncate
+                plen = packet_length_bytes
+            }
             if (cos value is not supported) {
                 cos = 0;
                 // Recommmended to log error about unsupported cos
                 // value.
             }
-            if (platform_port_valid(ep)) {
-                create a clone of the packet and send it to the packet
-                    buffer with the egress_port ep and
-                    class_of_service cos, after which it will start
-                    egress processing.  It will contain at most the
-                    first plen bytes of the packet as sent out from
-                    the egress deparser if trunc is true, otherwise
-                    the entire packet.
-            } else {
-                // Do not create a clone.  Recommended to log error
-                // about unsupported ep value.
+            for each pair (egress_port, instance) in the set {
+                Create a clone of the packet and send it to the packet
+                buffer with the egress_port, instance, and
+                class_of_service cos, after which it will start egress
+                processing.  It will contain at most the first plen
+                bytes of the packet as sent out from the egress
+                deparser if trunc is true, otherwise the entire
+                packet.
```

Comment:
Get rid of the platform_port_valid(ep) check -- assume that the control plane software will not program into the data plane invalid egress_port values for clone session replication sets.